aiohttp/cookiejar.py

```diff
@@ -194,15 +195,17 @@ def update_cookies(self,
 
         self._do_expiration()
 
-    def filter_cookies(self, request_url: URL=URL()) -> 'BaseCookie[str]':
+    def filter_cookies(self,
+                       request_url: URL=URL()
+                       ) -> Union['BaseCookie[str]', 'SimpleCookie[str]']:
         """Returns this jar's cookies filtered by their attributes."""
         self._do_expiration()
         if not isinstance(request_url, URL):
             warnings.warn("The method accepts yarl.URL instances only, got {}"
                           .format(type(request_url)),
                           DeprecationWarning)
             request_url = URL(request_url)
-        filtered = SimpleCookie()  # type: SimpleCookie[str]
+        filtered = SimpleCookie() if self._quote_cookie else BaseCookie()  # type: Union[SimpleCookie[str], BaseCookie[str]] # noqa: E501
```

Comment:
I'm pretty sure that the linter has a long line setting so that it's not violated. Adding noqa contradicts that intent.

Reply:
I struggled with it due to MyPy errors and never having used MyPy. See my own resolved thread above. 

I’ll have another look though.

Reply:
I've updated this. The MyPy error (`error: Name 'filtered' already defined on line 210` ) was about type annotating the variable more than once, not about defining the variable more than once.

The solution I've implemented (and personally think is cleaner) is this:
```python
        filtered: Union[SimpleCookie[str], BaseCookie[str]]
        if self._quote_cookie:
            filtered = SimpleCookie()
        else:
            filtered = BaseCookie()
```

but MyPy also allows this:
```python
        if self._quote_cookie:
            filtered = SimpleCookie()  # type: Union[SimpleCookie[str], BaseCookie[str]]
        else:
            filtered = BaseCookie()
```

Would you like me to change to the second solution, or are you fine with the one I implemented, @webknjaz ?